### ISSUE TITLE
Dependency vs Dependencies

### ISSUE THREAD

**Issue by Ivy8127** [ModECI/MDF#285]
Hi, @kmantel, with conditions on graph-scheduler, I noticed a difference in function arguments. There is the use of dependencies vs dependency, both of them work, which argument should be used uniformly across the board?

I've also been trying to run Threshold conditions locally, I started with the simple example you had in a [Recent PR](https://github.com/ModECI/MDF/pull/266/files#diff-7cab305a6a78540131fcce1fafb912ccf6b52aa50b4a0a2b794aea71004d5a95R180). However, this error is getting thrown whenever I create a termination condition with a threshold condition.

```raise ValueError(
ValueError: No param_A evaluable object for Node(metadata=None, id='A', input_ports=[InputPort(metadata=None, id='input_port1', shape=None, type=None)], 
functions=[], parameters=[Parameter(metadata=None, id='param_A', value='param_A + 1', default_initial_value=0, time_derivative=None, function=None, args=None, conditions=[])], 
output_ports=[OutputPort(metadata=None, id='output_1', value='param_A', shape=None, type=None)]), options: ['input_port1', 'param_A', 'output_1']```

**Comment by kmantel** [comment]
For conditions outside of mdf.Condition, `dependencies` is used when there are multiple accepted arguments as in [Any](https://kmantel.github.io/graph-scheduler/Condition.html#graph_scheduler.condition.Any), while `dependency` is used when there is only one argument as in [Threshold](https://kmantel.github.io/graph-scheduler/Condition.html#graph_scheduler.condition.Threshold). But for mdf.Conditions, there was a preference to use `dependencies` everywhere. In order not to break the interface in graph-scheduler, I added a parser to change  `dependencies` to `dependency` internally where necessary: https://github.com/ModECI/MDF/blob/8a89c2f8356c5c982e9c287f9a5b521eaaf06b8d/src/modeci_mdf/execution_engine.py#L1183-L1195

The error should be fixed now on main as of #280, that failure was hidden during merge due to CI issues

**Comment by Ivy8127** [comment]
> For conditions outside of mdf.Condition, `dependencies` is used when there are multiple accepted arguments as in [Any](https://kmantel.github.io/graph-scheduler/Condition.html#graph_scheduler.condition.Any), while `dependency` is used when there is only one argument as in [Threshold](https://kmantel.github.io/graph-scheduler/Condition.html#graph_scheduler.condition.Threshold). But for mdf.Conditions, there was a preference to use `dependencies` everywhere. In order not to break the interface in graph-scheduler, I added a parser to change `dependencies` to `dependency` internally where necessary:
> 
> https://github.com/ModECI/MDF/blob/8a89c2f8356c5c982e9c287f9a5b521eaaf06b8d/src/modeci_mdf/execution_engine.py#L1183-L1195
> 
> The ValueError with Threshold should be fixed now on main as of #280, that failure was hidden during merge due to CI issues

This clears up the confusion on my part. @pgleeson
